hs_composite_resource/tests/test_composite_resource_user_zone.py

```diff
@@ -205,3 +1005,55 @@ def _create_composite_resource(self):
             move=False,
             metadata=[]
         )
+
+    def _create_generic_aggregation(self):
+        # make sure composite_resource is created in federated user zone
+        fed_path = '/{zone}/home/{user}'.format(zone=settings.HS_USER_IRODS_ZONE,
+                                                user=settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+        self.assertEqual(self.composite_resource.resource_federation_path, fed_path)
+        res_file = self.composite_resource.files.first()
+        # create the generic aggregation (logical file)
+        GenericLogicalFile.set_file_type(self.composite_resource, self.user, res_file.id)
+
+    def _create_raster_aggregation(self):
+        # make sure composite_resource is created in federated user zone
+        fed_path = '/{zone}/home/{user}'.format(zone=settings.HS_USER_IRODS_ZONE,
+                                                user=settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+        self.assertEqual(self.composite_resource.resource_federation_path, fed_path)
+        res_file = self.composite_resource.files.first()
+        # create the raster aggregation (logical file)
+        if res_file.file_folder is None:
+            GeoRasterLogicalFile.set_file_type(self.composite_resource, self.user,
+                                               file_id=res_file.id)
+        else:
+            GeoRasterLogicalFile.set_file_type(self.composite_resource, self.user,
+                                               folder_path=res_file.file_folder)
+
+    def _test_copy_or_version_of_composite_resource(self, aggr_type,  action):
+        self._create_composite_resource()
+        self.assertEqual(CompositeResource.objects.all().count(), 1)
+        # create the specified aggregation (logical file)
+        if aggr_type == GenericLogicalFile:
```

Comment:
I feel like the aggregation should be created before calling this method.

Reply:
Calling which method?

Reply:
I was referring to pulling this if/else block out of this method and just calling the create method before _test_copy_or_version_of_composite_resource()

it's not a big deal, more of my own preference for organization.